woocommerce/woocommerce-ios
# [Store onboarding] Setting to show/hide the list

Part of: #9561


- [ ] ⚠️ Please review https://github.com/woocommerce/woocommerce-ios/pull/9572 as well

## Description

We want to allow the user to hide/dismiss the Store onboarding task list.

This PR,

- Adds a setting to hide or show the onboarding list from the settings screen.
- Adds tracking event.

## Testing instructions

- Enable `hideStoreOnboardingTaskList` feature flag by returning `true` from `DefaultFeatureFlagService`
- Login into a store which has pending store onboarding tasks
- Tap `Menu` -> `Settings`
- Toggle the `Store Setup List` switch and validate that the store onboarding task list in the dashboard is shown/hidden accordingly. 
- Hiding the list from the dashboard by tapping `...` -> `Hide store setup list` should update the settings screen as well.

**Tracks**

Settings
- Tap `Menu` -> `Settings`
- Toggle the `Store Setup List` switch from ON to OFF.
- Validate that the following event is tracked.

```
Tracked store_onboarding_hide_list, properties: [AnyHashable("blog_id"): 218298555, AnyHashable("is_wpcom_store"): false, AnyHashable("pending_tasks"): "payments,store_details", AnyHashable("source"): "settings"]
```

Dashboard
- From the dashboard hide the task list by tapping `...` -> `Hide store setup list`
- Validate that the following event is tracked.

```
Tracked store_onboarding_hide_list, properties: [AnyHashable("pending_tasks"): "payments,store_details", AnyHashable("is_wpcom_store"): false, AnyHashable("blog_id"): 218298555, AnyHashable("source"): "onboarding_list"]
```

Validate that the  `pending_tasks` value matches the pending onboarding tasks. 

## Screenshots

![Simulator Screen Recording - iPhone 14 - 2023-04-28 at 07 38 08](https://user-images.githubusercontent.com/524475/235037395-1eb6ecd1-ad57-487f-9827-28a89504fbed.gif)

---
